src/shared/utils/diagnose.ts

```diff
@@ -149,9 +165,102 @@ const diagnoseV3 = ({ kind, document, debug, mode }: { kind: Kind; document: any
 
   try {
     // eslint-disable-next-line @typescript-eslint/no-unused-expressions
-    mode === "strict"
-      ? VerifiableCredentialWrappedProofStrict.check(document.proof)
-      : VerifiableCredentialWrappedProof.check(document.proof);
+    mode === "strict" ? WrappedProofStrictV3.check(document.proof) : WrappedProofV3.check(document.proof);
+  } catch (e) {
+    if (e instanceof Error) {
+      return handleError(debug, e.message);
+    } else {
+      console.error(e);
+    }
+  }
+
+  if (kind === "signed") {
+    if (!document.proof) {
+      return handleError(debug, `The document does not have a proof`);
+    }
+    try {
+      SignedWrappedProofV3.check(document.proof);
+    } catch (e) {
+      if (e instanceof Error) {
+        return handleError(debug, e.message);
+      } else {
+        console.error(e);
+      }
+    }
+  }
+  return [];
+};
+
+const diagnoseV4 = ({
+  kind,
+  document,
+  debug,
+  mode,
+}: {
+  kind: Exclude<Kind, "raw">;
+  document: any;
+  debug: boolean;
+  mode: Mode;
+}) => {
+  /* Wrapped document checks */
+  try {
+    // 1. Since OA v4 has deprecated a few properties from v2/v3, check that they are not used
+    const deprecatedProperties = ["version", "openAttestationMetadata"];
+    const documentProperties = Object.keys(document);
+    const deprecatedDocumentProperties = documentProperties.filter((p) => deprecatedProperties.includes(p));
+
+    if (deprecatedDocumentProperties.length > 0) {
+      return handleError(
+        debug,
+        `The document has outdated properties previously used in v2/v3. The following properties are no longer in use in a v4 document: ${deprecatedDocumentProperties}`
+      );
+    }
+
+    // 2. Ensure that required @contexts are present
+    // @context: [Base, OA, ...]
+    const contexts = ["https://www.w3.org/2018/credentials/v1", ContextUrl.v4_alpha];
+    if (isStringArray(document["@context"])) {
```

Comment:
can we say as of today, to differentiate OA v4 version, this `ContextUrl.v4_alpha` context url is the main deal breaker?

Reply:
Yes, that seems to be the way at the moment because the `version` property has been omitted. To be exact, can check against `@context` and `type`